packages/utils/src/lib/reports/generate-md-report.ts

```diff
@@ -1,171 +1,150 @@
-import { AuditReport, Issue, Report, Table } from '@code-pushup/models';
+import { InlineText, MarkdownDocument, md } from 'build-md';
+import { AuditReport, Issue, Report } from '@code-pushup/models';
 import { formatDate, formatDuration } from '../formatting';
-import { SPACE, html, md } from '../text-formats';
-import {
-  FOOTER_PREFIX,
-  README_LINK,
-  issuesTableHeadings,
-  reportHeadlineText,
-} from './constants';
+import { HIERARCHY } from '../text-formats';
+import { FOOTER_PREFIX, README_LINK, reportHeadlineText } from './constants';
 import { metaDescription, tableSection } from './formatting';
 import {
   categoriesDetailsSection,
   categoriesOverviewSection,
 } from './generate-md-report-categoy-section';
 import { ScoredReport } from './types';
-import {
-  formatReportScore,
-  getPluginNameFromSlug,
-  scoreMarker,
-  severityMarker,
-} from './utils';
-
-const { h1, h2, h3, lines, link, section, code: codeMd } = md;
-const { bold: boldHtml, details } = html;
+import { formatReportScore, scoreMarker, severityMarker } from './utils';
 
 export function auditDetailsAuditValue({
   score,
   value,
   displayValue,
-}: AuditReport) {
-  return `${scoreMarker(score, 'square')} ${boldHtml(
+}: AuditReport): InlineText {
+  return md`${scoreMarker(score, 'square')} ${md.bold(
     String(displayValue ?? value),
   )} (score: ${formatReportScore(score)})`;
 }
 
 export function generateMdReport(report: ScoredReport): string {
-  const printCategories = report.categories.length > 0;
-
-  return lines(
-    h1(reportHeadlineText),
-    printCategories ? categoriesOverviewSection(report) : '',
-    printCategories ? categoriesDetailsSection(report) : '',
-    auditsSection(report),
-    aboutSection(report),
-    `${FOOTER_PREFIX}${SPACE}${link(README_LINK, 'Code PushUp')}`,
-  );
+  return new MarkdownDocument()
+    .heading(HIERARCHY.level_1, reportHeadlineText)
+    .$if(report.categories.length > 0, doc =>
+      doc.$concat(
+        categoriesOverviewSection(report),
+        categoriesDetailsSection(report),
+      ),
+    )
+    .$concat(auditsSection(report), aboutSection(report))
+    .rule()
```

Comment:
❔ What does `rule` mean?

Reply:
It's a [horizontal rule](https://www.markdownguide.org/basic-syntax/#horizontal-rules). It wasn't there before, but I though it made sense to separate the footer this way.